packages/ui/src/components/field-enabled.tsx

```diff
@@ -0,0 +1,39 @@
+import { useClerk } from '@clerk/clerk-react';
+import type { Attribute, EnvironmentResource } from '@clerk/types';
+
+type NegatedAttribute = `!${Attribute}`;
+type PickOption = Attribute | NegatedAttribute;
+
+export function FieldEnabled({
```

Comment:
I know we talked about this conditional rendering component and in practice it felt nice, it quickly can become more difficult when needing to check multiple fields + required status. In my PR I ultimately remove the component approach and just accessed the data directly in the component here https://github.com/clerk/javascript/blob/main/packages/ui/src/components/sign-up/sign-up.tsx#L36-L42

I also moved into a sub component `<SignUpComponentLoaded />` so that we can access the data without having to check if clerk is defined since the Root components handle that for us. https://github.com/clerk/javascript/blob/main/packages/ui/src/components/sign-up/sign-up.tsx#L25

Theres likely an abstraction we can do to make accessing this data a bit cleaner but I opted for keeping it simple for now and see how you access similar data in SignIn component.